### PR TITLE
feat(cli): add autocompletions (RDT-810)

### PR DESCRIPTION
**Pull request by xastap00**
Closes https://github.com/espressif/idf-build-apps/issues/27

### CODE REVIEW

File: idf_build_apps/constants.py [hunk 1]
```diff
@@ -95,3 +95,41 @@ class BuildStage(str, enum.Enum):
...
+
+    autoload -U compinit
+    compinit
+
+    Afterwards you can enable completions for idf-build-apps:
+
+    eval "$(register-python-argcomplete idf-build-apps)"
+
+    NOTE: If your version of argcomplete is earlier than v3, you may need to
+    have bashcompinit enabled in zsh by running:
```
Comment (hfudev) [inline idf_build_apps/constants.py:127]: let's add `argcomplete>=3` in our dependency list to simplify it.

File: idf_build_apps/main.py [hunk 2]
```diff
@@ -674,14 +675,34 @@ def get_parser() -> argparse.ArgumentParser:
...
+        'Run `idf-build-apps complations --help` for more information on a command',
+        description='Without argument print instructions for manual activation. '
+        'With argument add the autocompletion activation script to the shell rc file for bash, zsh, and fish. '
+        'Other shells are not supported',
+        formatter_class=IdfBuildAppsCliFormatter,
+    )
+    completions_parser.add_argument(
+        '-a', '--activate', action='store_true', help='Activate autocompletion automatically and permanently'
+    )
+
```
Comment (hfudev) [inline idf_build_apps/main.py:690]: I recommend adding one more argument here, 

```python
'--shell', choices=['auto', 'bash', 'zsh', 'fish'], default='auto'
```

so users could specify something if `$SHELL` not exists.

**Comment by xastap00** [comment]
Help messages for the `find` and `build` commands are also added

**Comment by xastap00** [comment]
@kumekay

**Comment by hfudev** [comment]
Thank you for the PR!

**Pull request merged**